demosplan/DemosPlanCoreBundle/Command/Data/DeleteProcedureCommand.php

```diff
@@ -0,0 +1,479 @@
+            $this->deactivateForeignKeyChecks();
+
+            // delete all statements and connected entities
+            $this->output->writeln('Deleting All Statements');
+            $this->processAllStatements();
+
+            // delete all annotated statement pdfs -> pages -> files
+            $this->output->writeln('Deleting Annotated PDFs');
+            $this->processAnnotatedStatementPdfs();
+
+            // delete procedure elements -> files
+            $this->output->writeln('Deleting Elements');
+            $this->processElements();
+
+            // Procedure Behavior Definition
+            $this->output->writeln('Deleting Behavior Definitions');
+            $this->deleteBehaviorDefinitions();
+
+            // Procedure UI Definition
+            $this->output->writeln('Deleting UI Definitions');
+            $this->deleteUiDefinitions();
+
+            // form definitions -> field definitions
+            $this->output->writeln('Deleting Form Definitions');
+            $this->processFormDefinitions();
+
+            // delete gis layers
+            $this->output->writeln('Deleting Gis Layers');
+            $this->processGisLayers();
+
+            // delete procedure news
+            $this->output->writeln('Deleting News');
+            $this->deleteProcedureNews();
+
+            // delete tag topics -> tags
+            $this->output->writeln('Deleting Tags');
+            $this->processTags();
+
+            // delete predefined text categories -> predefined texts
+            $this->output->writeln('Deleting Predefined Texts');
+            $this->processPredefinedTexts();
+
+            // delete draft statements
+            $this->output->writeln('Deleting Draft Statements');
+            $this->deleteDraftStatements();
+
+            // delete import_emails -> attachments
+            $this->output->writeln('Deleting Import Emails');
+            $this->processImportEmails();
+
+            // delete hashed queries
+            $this->output->writeln('Deleting hashed Queries');
+            $this->deleteHashedQueries();
+
+            // delete remaining procedure files
+            $this->output->writeln('Deleting Procedure Files');
+            $this->deleteFromTableByIdentifierArray('_files', 'procedure_id', [$this->procedureId]);
+
+            // delete procedure report entries
+            $this->output->writeln('Deleting Report Entries');
+            $this->deleteReportEntriesByIdentifierAndType('procedure', [$this->procedureId]);
+
+            // delete procedure itself
+            $this->output->writeln('Deleting Procedure');
+            $this->deleteProcedure();
+
+            // reactivate foreign key checks
+            $this->output->writeln('Activate FK Checks');
+            $this->activateForeignKeyChecks();
+
+            $this->output->writeln("Procedure $this->procedureId was purged successfully!");
+
+            return Command::SUCCESS;
```

Comment:
When not in dry-run you should repopulate elasticsearch

Reply:
you should probably pass the environment, as the command will be performed at prod servers, where no dev environment is available

Reply:
Done in [5160eff](https://github.com/demos-europe/demosplan-core/pull/1958/commits/5160effd139a87571a03d6aac785795342c4f215)